site/docs/manual/Architecture.md

```diff
@@ -515,11 +517,41 @@ class SystemName extends System {
 
 If there is a `reactive query` (A query that *listens* for entities added or removed to it or which components has changed, [more info](/manual/Architecture?id=reactive-queries)) on the list of queries defined by a system, this system is called `reactive system` as it will react to changes on the entities and its components.
 
+It's important to notice that if you plan to mutate the results of a query while you are iterating it (eg: adding or removing components that will make not match the query structure anymore, or removing the entity itself) you should traverse the results in reverse order:
+```javascript
+let results = this.queries.queryA.results;
+for (var i = 0; i < results.length; i++) {
+  let entity = results[i];
+  if (i === 1) {
+    // This will cause the results list to mutate and results.length will be decremented and you won't reach the end elements in there.
+    entity.remove();
+  }
+}
+
+// The correct way to do it
+let results = this.queries.queryA.results;
+for (var i = 0; i < results.length; i++) {
```

Comment:
Shouldn't we be decrementing here?

Reply:
Actually yes ;D that's the main point actualy of this hehe changing it